File: mocker-front/src/app/core/service/generate-type.service.ts
import { Injectable } from '@angular/core';
import { GenerateTypeModel } from '@core/domain/model/entity/generate-type.model';
import { AbstractService } from '@core/service/abstract.service';

/**
 * @author Do Quoc Viet
 */

@Injectable({
  providedIn: 'root'
})
export class GenerateTypeService extends AbstractService<GenerateTypeModel> {
  override ENTITY_URL: string = 'generate-type';
}

File: mocker-back/src/main/java/com/mocker/service/GenerateTypeService.java
package com.mocker.service;

import com.mocker.domain.model.entity.GenerateType;

import java.util.List;
import java.util.UUID;

/**
 * @author Luong Quoc Trung, Do Quoc Viet
 */

public interface GenerateTypeService {

    /**
     * GENERATETYPE.01 GET /generate-type/{id}
     *
     * @param id of generateType
     * @return generateType
     */
    GenerateType getGenerateType(UUID id);

    /**
     * GENERATETYPE.02 GET /generate-type/{typeId}
     *
     * @param sqlTypeId of sqlType
     * @return generateType by sqlTypeId
     */
    List<GenerateType> getGenerateTypesBySQLType(UUID sqlTypeId);

    /**
     * GENERATETYPE.03 GET /generate-type
     *
     * @return generateTypes
     */
    List<GenerateType> getGenerateTypes();
}
